domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/domain/topic/domain/TopicWorkflowImplementation.kt

```diff
@@ -0,0 +1,22 @@
+package com.krzykrucz.elesson.currentlesson.domain.topic.domain
+
+import arrow.core.getOrElse
+import com.krzykrucz.elesson.currentlesson.domain.NaturalNumber
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.LessonIdentifier
+
+fun chooseTopic(): ChooseTopic = { topicTitle, finishedLessonsCount, checkedAttendance ->
+    val attendance = checkedAttendance.attendance
+    InProgressLesson(
+            LessonIdentifier(
+                    date = attendance.date,
+                    lessonHourNumber = attendance.lessonHourNumber,
+                    className = attendance.className
+            ),
+            LessonTopic(LessonOrdinalNumber(calculateLessonOrdinalNumber(finishedLessonsCount)), topicTitle, attendance.date),
+            checkedAttendance
+    )
+}
+
+private fun calculateLessonOrdinalNumber(finishedLessonsCount: FinishedLessonsCount) =
```

Comment:
looking at the place where and how it used, maybe `FinishedLessonsCount` should be just a dependency, I mean another function to provide below behaviour because of this function `chooseTopic` not only creates `InProgressLesson` but also calculates `LessonOrdinalNumber`